wdl-format/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+use comments::format_preceding_comments;
+use formatter::Formatter;
+
+/// Newline constant used for formatting on windows platforms.
+#[cfg(windows)]
+pub const NEWLINE: &str = "\r\n";
+/// Newline constant used for formatting on non-windows platforms.
+#[cfg(not(windows))]
+pub const NEWLINE: &str = "\n";
+/// String terminator constant used for formatting.
+const STRING_TERMINATOR: char = '"';
+
+/// A trait for elements that can be formatted.
+pub trait Formattable {
+    /// Format the element and write it to the writer.
+    fn format<T: std::fmt::Write>(
+        &self,
+        writer: &mut T,
+        formatter: &mut Formatter,
+    ) -> std::fmt::Result;
+}
+
+impl Formattable for Version {
+    fn format<T: std::fmt::Write>(
+        &self,
+        writer: &mut T,
+        _formatter: &mut Formatter,
+    ) -> std::fmt::Result {
+        write!(writer, "{}", self.as_str())
+    }
+}
+
+impl Formattable for VersionStatement {
+    fn format<T: std::fmt::Write>(
+        &self,
+        writer: &mut T,
+        formatter: &mut Formatter,
+    ) -> std::fmt::Result {
+        let mut preamble_comments = Vec::new();
+        let mut lint_directives = Vec::new();
+        let comment_buffer = &mut String::new();
+        for sibling in self.syntax().siblings_with_tokens(Direction::Prev).skip(1) {
+            match sibling.kind() {
+                SyntaxKind::Comment => {
+                    let comment = Comment::cast(
+                        sibling
+                            .as_token()
+                            .expect("Comment should be a token")
+                            .clone(),
+                    )
+                    .expect("Comment should cast to a comment");
+                    comment.format(comment_buffer, formatter)?;
+                    if comment_buffer.starts_with("#@") {
+                        lint_directives.push(comment_buffer.clone());
+                    } else {
+                        preamble_comments.push(comment_buffer.clone());
+                    }
+                    comment_buffer.clear();
+                }
+                SyntaxKind::Whitespace => {
+                    // Ignore
+                }
+                _ => {
+                    unreachable!("Unexpected syntax kind: {:?}", sibling.kind());
+                }
+            }
+        }
+
+        for comment in preamble_comments.iter().rev() {
+            write!(writer, "{}", comment)?;
+        }
+
+        // If there are preamble comments, ensure a blank line is inserted
+        if !preamble_comments.is_empty() {
+            write!(writer, "{}", NEWLINE)?;
+        }
+
+        for comment in lint_directives.iter().rev() {
+            write!(writer, "{}", comment)?;
+        }
+
+        let version_keyword = first_child_of_kind(self.syntax(), SyntaxKind::VersionKeyword);
+        write!(writer, "{}", version_keyword)?;
+        format_inline_comment(&version_keyword, writer, formatter, true)?;
+
```

Comment:
`state` in these cases needs to be updated to `formatter`.

Reply:
Was this comment made on an outdated commit? I don't see `state` anywhere in the snippet shown.